crates/ruff_linter/src/rules/flake8_todos/rules/todos.rs

```diff
@@ -222,11 +223,14 @@ impl Violation for MissingSpaceAfterTodoColon {
     }
 }
 
+// TD003 will trigger if neither the TODO itself nor the following line
+// matches any of the patterns.
 static ISSUE_LINK_REGEX_SET: Lazy<RegexSet> = Lazy::new(|| {
     RegexSet::new([
-        r"^#\s*(http|https)://.*", // issue link
-        r"^#\s*\d+$",              // issue code - like "003"
-        r"^#\s*[A-Z]{1,6}\-?\d+$", // issue code - like "TD003"
+        r"^#.*(http|https)://.*", // issue link
+        r"^#.*#\d+",              // issue code - like "#003"
+        r"^#\s\d+",               // issue code - like "003"
+        r"^#\s[A-Z]{1,6}\-?\d+",  // issue code - like "TD003"
```

Comment:
Is it correct that the reason you removed the `\s` from the "issue link" regex is so that the link can be anywhere in the comment?

Also, can we keep the asterisk for `\s*` so that there can be any number of whitespace instead of just one? Is there a reason you removed them?

Reply:
> Is it correct that the reason you removed the \s from the "issue link" regex is so that the link can be anywhere in the comment?

Yep, absolutely! 

> Also, can we keep the asterisk for \s* so that there can be any number of whitespace instead of just one? Is there a reason you removed them?

I removed 'em because I'd consider `#     TODO` incorrect, but if we were to enforce that, it should be a separate rule. I've re-added them.